lib/curl_req.ex

```diff
@@ -79,30 +85,97 @@ defmodule CurlReq do
         body -> ["-d", body]
       end
 
+    redirect =
+      case req.options do
+        %{redirect: true} -> ["-L"]
+        _ -> []
+      end
+
     method =
       case req.method do
         nil -> ["-X", "GET"]
+        :head -> ["-I"]
         m -> ["-X", String.upcase(to_string(m))]
       end
 
     url = [to_string(req.url)]
 
-    CurlReq.Shell.cmd_to_string("curl", headers ++ cookies ++ body ++ method ++ url)
+    CurlReq.Shell.cmd_to_string(
+      "curl",
+      headers ++ cookies ++ body ++ method ++ redirect ++ auth ++ url
+    )
   end
 
+  defp maybe_remove_auth_header(headers, []), do: headers
```

Comment:
I was confused, but now I understand how this is supposed to work. This second option means we remove the auth header only when we have found basic auth in req.options. It would be nice to move these bits together though, so it's more obvious what is happening here.